### PR TITLE
Porting Steady-State tutorial for Optomechanical Systems to QuTiP 5

### PR DESCRIPTION
**Pull request by obliviateandsurrender**
Porting the demo `Steady-State: Optomechanical System in the Single-Photon Strong-Coupling Regime` tutorial to QuTiP 5 (v5) as noted in #92 

1. Added description of new solvers and methods based on `v5` [docstring](https://qutip.readthedocs.io/en/qutip-5.0.x/apidoc/functions.html#qutip.solver.steadystate.steadystate).
2. Adapted the steady-state solution code for `v5`.
3. Adapt for accessing `ndarrays` for the `Dense` representation of `Qobj`. 
4. Change an argument name for `plt.spy` visualization. 
5. Add an assertion for non-zero elements for sanity check.

### CODE REVIEW

**Comment by hodgestar** [comment]
Thank you for taking a look. Could you try make some better assertions on the results and move them into ` Testing` section at the end? See the other tutorials for examples (some have better checks than others).

I've activated CI. If the CI run produces any errors, please update the PR to fix them.

File: tutorials-v5/time-evolution/019_optomechanical-steadystate.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+plt.colorbar(c, ax=ax);
+```
+
+## About
+
+```python
+about()
+```
+
+### Testing
```
Comment (hodgestar) [inline tutorials-v5/time-evolution/019_optomechanical-steadystate.md:256]:
```suggestion
## Testing
```
The about and testing headings should be at the same level.

File: tutorials-v5/time-evolution/019_optomechanical-steadystate.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+rho0 = psi0 @ psi0.dag()
+tlist = np.linspace(0, 1500, 1500)
+rho_evolve = mesolve(H, rho0, tlist, c_ops)
+rho_final = ptrace(rho_evolve.states[-1], 1)
+assert fidelity(rho_mech, rho_final) > 0.99
+
+# assert magnitude of diagonal elements are more than non-diagonal
+rho_ss_dmat = rho_ss.data.to_array()
+rho_ss_diag = np.diag(rho_ss.diag())
+assert np.all(np.abs(rho_ss_diag) - np.abs(rho_ss_dmat) <= 0)
```
Comment (hodgestar) [inline tutorials-v5/time-evolution/019_optomechanical-steadystate.md:270]: This seems true for any matrix -- i.e. that `abs(just the diagonals) - abs(full matrix)` has entries `<= 0`. Let's remove this block and think of a better check to add instead.

Comment (obliviateandsurrender) [reply tutorials-v5/time-evolution/019_optomechanical-steadystate.md]: Nice catch, I changed this test to now properly assert diagonal dominance.

**Comment by obliviateandsurrender** [comment]
I don't think the current build error is related to my changes, any ideas on fixing it? 🤔

**Comment by ArturDomingues** [comment]
See what I did in #98 this could probably help

File: tutorials-v5/time-evolution/013_nonmarkovian_monte_carlo.md [hunk 3]
```diff
@@ -203,7 +203,7 @@ We specify some numerical parameters, i.e., the number of trajectories and wheth
 nmmc_options = {"map": "parallel",
                 "norm_steps": 10}  # options specific to nm_mcsolve
 options = {"progress_bar": "enhanced"}  # options shared by all solvers
-ntraj = 5000
+ntraj = 5500
```
Comment (obliviateandsurrender) [inline tutorials-v5/time-evolution/013_nonmarkovian_monte_carlo.md:206]: Just to increase the robustness of the tests for this, it failed in the last CI run.

File: tutorials-v5/miscellaneous/JAX_backend.md [hunk 4]
```diff
@@ -119,7 +119,10 @@ def final_expect(solver, rho0, t, w):
 dfinal_expect_dt = jax.jit(
     jax.grad(final_expect, argnums=[2]), static_argnames=["solver"]
 )
-dfinal_expect_dt(solver, qutip.basis(10, 8, dtype="jax"), 0.1, 1.0)
+
+# TODO: use dfinal_expect_dt instead of final_expect when qutip-jax bug-fix
+# dfinal_expect_dt(solver, qutip.basis(10, 8, dtype="jax"), 0.1, 1.0)
+jax.grad(final_expect, argnums=[2])(solver, qutip.basis(10, 8, dtype="jax"), 0.1, 1.0)
```
Comment (obliviateandsurrender) [inline tutorials-v5/miscellaneous/JAX_backend.md:125]: Can revert this back if required!

**Comment by obliviateandsurrender** [comment]
@hodgestar, let me know if this needs more modifications!

File: tutorials-v5/time-evolution/013_nonmarkovian_monte_carlo.md (hunk 3 above)
Comment (hodgestar) [reply tutorials-v5/time-evolution/013_nonmarkovian_monte_carlo.md:207]: Thanks!

**Review by hodgestar — COMMENTED** [review]
Thanks @obliviateandsurrender.

**Pull request merged**